src/content/post/20170213_nodemcu_espruino_step_by_step.md

```diff
@@ -0,0 +1,134 @@
+Make sure to use the firmware compatible with ESP8266.
+
+### Flash the espruino firmware
+To set NodeMCU to flash mode, keep pressed the "Flash" button while plugging in the device into the USB.
+
+Before we actually upload the new firmware, we have to make sure everything is clean. For that, we first erase the flash on the device by executing: 
+```
+esptool.py --port /dev/ttyUSB0 erase_flash
+```
+
+After that we can upload the Espruino firmware with the following command:
+```
+esptool.py --port /dev/ttyUSB0 --baud 115200 write_flash --flash_freq 80m --flash_mode qio --flash_size 32m 0x0000 "boot_v1.6.bin" 0x1000 espruino_esp8266_user1.bin 0x3FC000 esp_init_data_default.bin 0x37E000 blank.bin
+```
+
+Please note that bauds sets are different from those listed on the device.
+You should find the same instruction in a README file within the downloaded firmware.
+
+### Espruino IDE
+[Espruino Web-IDE plugin for Chrome](https://chrome.google.com/webstore/detail/espruino-web-ide/bleoifhkdalbjfbobjackfdifdneehpo?hl=en) is the easiest tool to use to upload custom code into the NodeMCU.
+Once installed, go to settings > communications and update Baud rate to 115200.
+
+Now if you click on the connect button, you should be able to get access to the device.
+
+### Examples
+
+#### Connect to wifi
+```
+var wifi = require("Wifi");
+wifi.setHostname("Sparkfabrik-espruino");
+wifi.connect("<YOUR-SSID-HERE>", {password:"<YOUR-PASSWORD-HERE>"}, function(err){
+ console.log("connected? err=", err, "info=", wifi.getIP());
+});
+wifi.stopAP();
+wifi.save();
+```
+#### Temp and RH data over internet
+NodeMCU GPIO label don't match the actual GPIO pin. With Espruino, there is a NodeMCU class that allows an easy mapping between Espruino and NodeMCU pins.
+Please refer to the [espruino reference](http://www.espruino.com/Reference#NodeMCU) for it.
+```
+pin=NodeMCU.D2;
+var dht11 = require("DHT11").connect(pin);
+var dht11_read_test = function() {
+  dht11.read(
+    function(a){
+      console.log("Temp is "+a.temp.toString()+ " and RH is "+a.rh.toString());
+      var data = {
+        temp: a.temp.toString(),
+        rh:a.rh.toString()
+      };
+      putDweet("sparkfabrik-espruino", data, function(response) {
+        console.log(response);
+      });
+     });
+};
+setInterval(dht11_read_test,60000);
+
+function putDweet(dweet_name, a, callback) {
+  var data = "";
+  for (var n in a) {
+    if (data.length) data+="&";
+    data += encodeURIComponent(n)+"="+encodeURIComponent(a[n]); 
+  }
+  var options = {
+    host: 'dweet.io',
+    port: '80',
+    path:'/dweet/for/'+dweet_name+"?"+data,
+    method:'POST'
+  };
+  require("http").request(options, function(res)  {
+    var d = "";
+    res.on('data', function(data) { d+=data; });
+    res.on('close', function(data) {
+      if (callback) callback(d);
+    });
+ }).end();
+}
+
+save();
+```
+
```

Comment:
I'd add a "conclusion" or "Final result" paragraph, so that the reader knows what to expect as outcome.

It is not just a "here is it" matter, I love articles that allows me to understand why I am undertaking the project in the first place, so setting the stake and describing the final state helps IMHO to define the value generated by that piece of knowledge.

I used a lot of words to say something really simple, I guess: elevate your pitch about _why_ and _what's the desired final state_ :)

Makes sense?